scripts/deploy/deployScoutProtocol.ts

```diff
@@ -0,0 +1,217 @@
+import { execSync } from 'node:child_process';
+import fs from 'node:fs';
+import path from 'node:path';
+
+import dotenv from 'dotenv';
+import { task } from 'hardhat/config';
+import inquirer from 'inquirer';
+import { createPublicClient, createWalletClient, encodeDeployData, http, isAddress, parseAbiItem } from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
+
+import { getConnectorFromHardhatRuntimeEnvironment, getConnectorKey } from '../../lib/connectors';
+
+dotenv.config();
+
+const PRIVATE_KEY = (
+  process.env.PRIVATE_KEY?.startsWith('0x') ? process.env.PRIVATE_KEY : `0x${process.env.PRIVATE_KEY}`
+) as `0x${string}`;
+
+task('deployScoutProtocol', 'Deploys or updates the ScoutProtocol contracts').setAction(async (taskArgs, hre) => {
```

Comment:
Same as buildernft

Deploy a new implementation

Point a proxy to it, or also deploy a new proxy